integration-test/987-national-forests.py

```diff
@@ -0,0 +1,205 @@
+
+# leisure=common in Blithedale Summit Open Space Preserve with
+# boundary=national_park and boundary:type=protected_area and protect_class=5
+# and operator=Marin County Parks should just be kind:common.
+#
+# https://www.openstreetmap.org/way/297452972
+assert_has_feature(
+    15, 5229, 12648, 'landuse',
+    { 'kind': 'common', 'id': 297452972 })
+
+# leisure=nature_reserve in Glen Canyon National Recreation Area with
+# boundary=national_park and boundary:type=protected_area and protect_class=5
+# and protection_title=National Recreation Area which I think should default
+# to kind:nature_reserve.
+#
+# http://www.openstreetmap.org/relation/5273153
+assert_has_feature(
+    12, 784, 1585, 'landuse',
+    { 'kind': 'nature_reserve', 'id': -5273153 })
+
+# leisure=nature_reserve in Parque Natural Sierra de Andújar with finally
+# kind:nature_reserve.
+#
+# https://www.openstreetmap.org/way/373769670
+assert_has_feature(
+    13, 4003, 3151, 'landuse',
+    { 'kind': 'nature_reserve', 'id': 373769670 })
+
+# boundary=protected_area in Naturpark Steigerwald with protect_class=5
+# should end up with kind: protected_area
+#
+# https://www.openstreetmap.org/relation/3875431
+assert_has_feature(
+    13, 4336, 2787, 'landuse',
+    { 'kind': 'protected_area', 'id': -3875431 })
+
+# boundary=national_park in Muir Woods National Monument with
+# leisure=nature_reserve and operator=National Park Service and
+# protect_class=3 should end up with kind:national_park
+#
+# https://www.openstreetmap.org/relation/6229828
+assert_has_feature(
+    16, 10452, 25302, 'landuse',
+    { 'kind': 'national_park', 'id': -6229828 })
+
+# leisure=park in Henry W. Coe State Park with boundary=protected_area and
+# protect_class=5 should end up with kind:park.
+#
+# https://www.openstreetmap.org/relation/318202
+assert_has_feature(
+    13, 1332, 3184, 'landuse',
+    { 'kind': 'park', 'id': -318202 })
+
+# operator=United States National Park Service and protect_class=2 in
+# Yosemite National Park with boundary=national_park and
+# leisure=nature_reserve should end up with kind:national_park.
+#
+# https://www.openstreetmap.org/relation/1643367
+assert_has_feature(
+    13, 1371, 3164, 'landuse',
+    { 'kind': 'national_park', 'id': -1643367 })
+
+# operator=United States National Park Service and protect_class=2 in Redwood
+# National Park with boundary=national_park and leisure=nature_reserve should
+# end up with kind:national_park.
+#
+# https://www.openstreetmap.org/relation/215231
+assert_has_feature(
+    13, 1274, 3066, 'landuse',
+    { 'kind': 'national_park', 'id': -215231 })
+
+# operator=United States National Park Service and protect_class=2 in
+# Yellowstone National Park with boundary=national_park and
+# leisure=nature_reserve should end up with kind:national_park.
+#
+# https://www.openstreetmap.org/relation/1453306
+assert_has_feature(
+    13, 1591, 2972, 'landuse',
+    { 'kind': 'national_park', 'id': -1453306 })
```

Comment:
To make Yellowstone work, Point Reyes looses out (until data is cleaned up). Works for me.